
 Makefile                 |   17 +++++------------
 scripts/kconfig/Makefile |   23 -----------------------
 2 files changed, 5 insertions(+), 35 deletions(-)

diff -Nurp linux-4.12.4-mga1/Makefile linux-4.12.4-mga1.mrp/Makefile
--- linux-4.12.4-mga1/Makefile	2017-08-03 21:40:56.164184146 +0300
+++ linux-4.12.4-mga1.mrp/Makefile	2017-08-03 21:42:11.538506065 +0300
@@ -557,9 +557,7 @@ ifeq ($(KBUILD_EXTMOD),)
 # Carefully list dependencies so we do not try to build scripts twice
 # in parallel
 PHONY += scripts
-scripts: scripts_basic include/config/auto.conf include/config/tristate.conf \
-	 asm-generic gcc-plugins
-	$(Q)$(MAKE) $(build)=$(@)
+scripts:
 
 # Objects we will link into vmlinux / subdirs we need to visit
 init-y		:= init/
@@ -1064,7 +1062,7 @@ prepare0: archprepare gcc-plugins
 	$(Q)$(MAKE) $(build)=.
 
 # All the preparing..
-prepare: prepare0 prepare-objtool
+prepare:
 
 PHONY += prepare-objtool
 prepare-objtool: $(objtool_target)
@@ -1287,13 +1285,8 @@ endif # CONFIG_MODULES
 CLEAN_DIRS  += $(MODVERDIR)
 
 # Directories & files removed with 'make mrproper'
-MRPROPER_DIRS  += include/config usr/include include/generated          \
-		  arch/*/include/generated .tmp_objdiff
-MRPROPER_FILES += .config .config.old .version .old_version \
-		  Module.symvers tags TAGS cscope* GPATH GTAGS GRTAGS GSYMS \
-		  signing_key.pem signing_key.priv signing_key.x509	\
-		  x509.genkey extra_certificates signing_key.x509.keyid	\
-		  signing_key.x509.signer vmlinux-gdb.py
+MRPROPER_DIRS  += ""
+MRPROPER_FILES += ""
 
 # clean - Delete most, but leave enough to build external modules
 #
@@ -1315,7 +1308,7 @@ clean: archclean vmlinuxclean
 #
 mrproper: rm-dirs  := $(wildcard $(MRPROPER_DIRS))
 mrproper: rm-files := $(wildcard $(MRPROPER_FILES))
-mrproper-dirs      := $(addprefix _mrproper_,Documentation/DocBook scripts)
+mrproper-dirs      := $(addprefix _mrproper_,Documentation/DocBook)
 
 PHONY += $(mrproper-dirs) mrproper archmrproper
 $(mrproper-dirs):
diff -Nurp linux-4.12.4-mga1/scripts/kconfig/Makefile linux-4.12.4-mga1.mrp/scripts/kconfig/Makefile
--- linux-4.12.4-mga1/scripts/kconfig/Makefile	2017-08-03 21:40:56.164184146 +0300
+++ linux-4.12.4-mga1.mrp/scripts/kconfig/Makefile	2017-08-03 21:42:11.538506065 +0300
@@ -19,25 +19,16 @@ endif
 unexport CONFIG_
 
 xconfig: $(obj)/qconf 3rdparty/Makefile
-	$< $(silent) $(Kconfig)
 
 gconfig: $(obj)/gconf 3rdparty/Makefile
-	$< $(silent) $(Kconfig)
 
 menuconfig: $(obj)/mconf 3rdparty/Makefile
-	$< $(silent) $(Kconfig)
 
 config: $(obj)/conf 3rdparty/Makefile
-	$< $(silent) --oldaskconfig $(Kconfig)
 
 nconfig: $(obj)/nconf 3rdparty/Makefile
-	$< $(silent) $(Kconfig)
 
 silentoldconfig: $(obj)/conf 3rdparty/Makefile
-	$(Q)mkdir -p include/config include/generated
-	$(Q)test -e include/generated/autoksyms.h || \
-	    touch   include/generated/autoksyms.h
-	$< $(silent) --$@ $(Kconfig)
 
 localyesconfig localmodconfig: $(obj)/streamline_config.pl $(obj)/conf 3rdparty/Makefile
 	$(Q)mkdir -p include/config include/generated
@@ -83,7 +74,6 @@ simple-targets := oldconfig allnoconfig
 PHONY += $(simple-targets)
 
 $(simple-targets): $(obj)/conf 3rdparty/Makefile
-	$< $(silent) --$@ $(Kconfig)
 
 PHONY += oldnoconfig savedefconfig defconfig
 
@@ -93,23 +83,10 @@ PHONY += oldnoconfig savedefconfig defco
 oldnoconfig: olddefconfig 3rdparty/Makefile
 
 savedefconfig: $(obj)/conf 3rdparty/Makefile
-	$< $(silent) --$@=defconfig $(Kconfig)
 
 defconfig: $(obj)/conf 3rdparty/Makefile
-ifeq ($(KBUILD_DEFCONFIG),)
-	$< $(silent) --defconfig $(Kconfig)
-else
-ifneq ($(wildcard $(srctree)/arch/$(SRCARCH)/configs/$(KBUILD_DEFCONFIG)),)
-	@$(kecho) "*** Default configuration is based on '$(KBUILD_DEFCONFIG)'"
-	$(Q)$< $(silent) --defconfig=arch/$(SRCARCH)/configs/$(KBUILD_DEFCONFIG) $(Kconfig)
-else
-	@$(kecho) "*** Default configuration is based on target '$(KBUILD_DEFCONFIG)'"
-	$(Q)$(MAKE) -f $(srctree)/Makefile $(KBUILD_DEFCONFIG)
-endif
-endif
 
 %_defconfig: $(obj)/conf 3rdparty/Makefile
-	$(Q)$< $(silent) --defconfig=arch/$(SRCARCH)/configs/$@ $(Kconfig)
 
 configfiles=$(wildcard $(srctree)/kernel/configs/$@ $(srctree)/arch/$(SRCARCH)/configs/$@)
 
